dashboard/db/migrate/20190808182650_rename_academic_year_workshop_names.rb

```diff
@@ -0,0 +1,49 @@
+class RenameAcademicYearWorkshopNames < ActiveRecord::Migration[5.0]
+  CSD_COURSE = 'CS Discoveries'.freeze
+  CSD_SUBJECTS = [
+    {form: '1-day Academic Year, Units 1 and 2', to: 'Workshop 1: Unit 3'},
+    {form: '1-day Academic Year, Unit 3', to: 'Workshop 2: Unit 4'},
+    {form: '1-day Academic Year, Units 4 and 5', to: 'Workshop 3: Unit 5'},
+    {form: '1-day Academic Year, Unit 6', to: 'Workshop 4: Unit 6'},
+    {form: '2-day Academic Year, Units 1 to 3', to: '2-day, Workshops 1+2: Units 3 and 4'},
+    {form: '2-day Academic Year, Units 4 to 6', to: '2-day, Workshops 3+4: Units 5 and 6'},
+  ].freeze
+
+  CSP_COURSE = 'CS Principles'.freeze
+  CSP_SUBJECTS = [
+    {from: '1-day Academic Year, Units 1 and 2', to: 'Workshop 1: Unit 3'},
+    {from: '1-day Academic Year, Unit 3', to: 'Workshop 2: Unit 4 and Explore Task'},
+    {from: '1-day Academic Year, Unit 4 + Explore Prep', to: 'Workshop 3: Unit 5 and Create Task'},
+    {from: '1-day Academic Year, Unit 5 + Create Prep', to: 'Workshop 4: Unit 5 and Multiple Choice Exam'},
+    {from: '2-day Academic Year, Units 1 to 3', to: '2-day, Workshops 1+2: Units 3-4 and Explore Task'},
+    {from: '2-day Academic Year, Units 4 and 5 + AP Prep', to: '2-day, Workshops 3+4: Unit 5, Create Task, and Multiple Choice Exam'},
+  ].freeze
+
+  def up
+    CSD_SUBJECTS.each do |subject_name|
+      Pd::Workshop.where(course: CSD_COURSE, subject: subject_name[:from]).
+        where(started_at: nil).
+        update_all(subject: subject_name[:to])
+    end
+
+    CSP_SUBJECTS.each do |subject_name|
+      Pd::Workshop.where(course: CSP_COURSE, subject: subject_name[:from]).
+        where(started_at: nil).
+        update_all(subject: subject_name[:to])
+    end
+  end
+
+  def down
+    CSD_SUBJECTS.each do |subject_name|
+      Pd::Workshop.where(course: CSD_COURSE, subject: subject_name[:to]).
+        where(started_at: nil).
```

Comment:
This isn't quite sound as a down migration, since a workshop might have started between running `up` and `down` and would therefore get missed by the `down` step.  In practice I think this is very unlikely to be an issue though.  Think we should worry about it?

Reply:
My assumption is `up` and `down` must be in the same transaction. If `up` fails then rails will automatically execute `down` to roll back. Since this is a write, it should have a lock on the table to prevent other from writing to it at the same time. This is just theory though, I'm not sure if Rails and MySQL do something different under the hood.

Reply:
In addition, this migration is less destructive than many other migrations such as removing column, renaming column, renaming table etc. I hope it's safe :)

Reply:
If `up` fails, Rails should not need to execute `down` because it will simply roll back the transaction.  That is to say, `up` and `down` won't necessarily happen in the same transaction.  `down` might run later if we find a bug post-deploy and have to [roll back a transaction](https://edgeguides.rubyonrails.org/active_record_migrations.html#rolling-back) prior to shipping a revert.

That said, if we don't have any active workshops of this type right now, I suspect we'll know things are stable before any of them open up.

Reply:
I see; I could use a more specific condition such as all workshops having the first session after a certain date.